File: GeneratedEntityReaderClasses/Components/InventoryGuiComponent.cs
namespace GeneratedNoitaClasses;
public class InventoryGuiComponent : NoitaComponentBase,  iNoitaType<InventoryGuiComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaBool has_opened_inventory_edit { get; set; } = new();
	public NoitaInt wallet_money_target { get; set; } = new();
	public NoitaBool mDisplayFireRateWaitBar { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		has_opened_inventory_edit.Read(s);
		wallet_money_target.Read(s);
		mDisplayFireRateWaitBar.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		has_opened_inventory_edit.Write(s);
		wallet_money_target.Write(s);
		mDisplayFireRateWaitBar.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/InventoryGuiComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class InventoryGuiComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    has_opened_inventory_edit:NoitaBool = field(default_factory=NoitaBool)
    wallet_money_target:NoitaInt = field(default_factory=NoitaInt)
    mDisplayFireRateWaitBar:NoitaBool = field(default_factory=NoitaBool)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.has_opened_inventory_edit.Read(s)
        self.wallet_money_target.Read(s)
        self.mDisplayFireRateWaitBar.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.has_opened_inventory_edit.Write(s)
        self.wallet_money_target.Write(s)
        self.mDisplayFireRateWaitBar.Write(s)

